sui_programmability/examples/games/tests/vdf_based_lottery_tests.move

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test_only]
+module games::vdf_based_lottery_tests {
+    use sui::test_scenario::{Self};
+    use games::vdf_based_lottery::{Self, Game, Ticket, GameWinner};
+
+    #[test]
+    #[expected_failure(abort_code = games::vdf_based_lottery::ESubmissionPhaseInProgress)]
+    fun test_complete_too_early() {
+        let user1 = @0x0;
+
+        let scenario_val = test_scenario::begin(user1);
+        let scenario = &mut scenario_val;
+
+        let clock = sui::clock::create_for_testing(test_scenario::ctx(scenario));
+
+        vdf_based_lottery::create(1000, 1000, &clock, test_scenario::ctx(scenario));
+        test_scenario::next_tx(scenario, user1);
+        let game_val = test_scenario::take_shared<Game>(scenario);
+        let game = &mut game_val;
```

Comment:
Same deal here as for `TestScenario`, just keep the game around, and you can call functions on it using receiver syntax, which will do the borrow automatically.